test/cases/cross-distro.sh

```diff
@@ -104,11 +104,26 @@ fi
 
 # ALL_DISTROS - all possible distros from upstream repository
 # ALL_EXPECTED_DISTROS - all distros matching host pattern
-# ALL_REMAINDERS - all the unrecognized distros
+# UNRECOGNIZED_DISTROS - all the unrecognized distros
 # Filter out beta and centos-stream, see GH issue #2257
 ALL_DISTROS=$(find "$REPO_PATH" -name '*.json' -printf '%P\n' | awk -F "." '{ print $1 }')
 ALL_EXPECTED_DISTROS=$(echo "$ALL_DISTROS" | grep -E "$PATTERN" | grep -Ev 'beta|stream' | sort)
-ALL_REMAINDERS=$(echo "$ALL_DISTROS" | grep -v "$RECOGNIZED_DISTROS")
+
+# Warning: filter out the remaining distros one by one instead of using grep b/c
+# the distro value rhel-93 will be matched by the grep pattern rhel-9 and will be
+# filtered out. If we're running on a RHEL 9.2 maching osbuild-composer doesn't
+# know anything about 9.3 images so the value rhel-9.3 should be treated as unrecognized
+# and error out as expected in the test snippet further below
+UNRECOGNIZED_DISTROS=""
```

Comment:
The `UNRECOGNIZED_DISTROS` variable is already defined on line no. 87.  I would suggest to prefix this one with `ALL_`, since its use is different.

Also wouldn't it be sufficient to match whole words or lines with `grep` instead? IOW use `$(echo "$ALL_DISTROS" | grep -vw "$RECOGNIZED_DISTROS")`... From what I tested, it seems to produce the expected results.

Reply:
> Also wouldn't it be sufficient to match whole words or lines with grep instead?

Probably. I didn't know about this option. Updated the PR accordingly.